massdriver-application-azure-vm-docker/main.tf

```
@@ -0,0 +1,152 @@
+locals {
+  split_vnet_id        = split("/", var.virtual_network_id)
+  vnet_name            = element(local.split_vnet_id, length(local.split_vnet_id) - 1)
+  vnet_resource_group  = element(local.split_vnet_id, index(local.split_vnet_id, "resourceGroups") + 1)
+  max_length           = 24
+  storage_account_name = substr(replace(var.name, "/[^a-z0-9]/", ""), 0, local.max_length)
+}
+
+module "application" {
+  source              = "github.com/massdriver-cloud/terraform-modules//massdriver-application?ref=ead16fe"
+  name                = var.name
+  service             = "vm"
+  resource_group_name = azurerm_resource_group.main.name
+  location            = azurerm_resource_group.main.location
+}
+
+resource "azurerm_resource_group" "main" {
+  name     = var.name
+  location = var.location
+  tags     = var.tags
+}
+
+resource "azurerm_linux_virtual_machine_scale_set" "main" {
+  name                            = var.name
+  resource_group_name             = azurerm_resource_group.main.name
+  location                        = azurerm_resource_group.main.location
+  disable_password_authentication = false
+  admin_username                  = "placeholder"
+  admin_password                  = random_password.main.result
+  # instances                       = var.auto_scaling.enabled ? var.scaleset.instances : 1
+  instances   = 1
+  sku         = var.machine_type
+  custom_data = base64encode(local.cloud_init_rendered)
+  # health_probe_id              = var.dns.enable_dns ? module.public_endpoint[0].azurerm_lb_probe : null
+  extension_operations_enabled = false
+  tags                         = var.tags
+
+  network_interface {
+    name    = var.name
+    primary = true
+    # network_security_group_id = var.dns.enable_dns ? module.public_endpoint[0].azurerm_network_security_group_id : null
+
+    dynamic "ip_configuration" {
+      for_each = var.dns.enable_dns ? [] : [1]
+      content {
+        name      = var.name
+        subnet_id = data.azurerm_subnet.default.id
```

```suggestion
        subnet_id = azurerm_subnet.main.id
```

Going off of my comment below, I think we want to use the subnet we're creating instead of the default subnet of the VNet bundle.